design/grammar.md

```diff
@@ -0,0 +1,13 @@
+# SeedBlock and SeedX Grammars
+
+## SeedBlock
+
+The syntax of SeedBlock is defined by an ANTLR v4 grammar file named [SeedBlock.g4](/grammars/SeedBlock.g4). The parser generated by ANTLR tool is used to parse the whole SeedBlock program or validate the partial user input during inline editing.
```

Comment:
Please wrap the text paragraphs and keep <= 80 chars per line. Per Google's markdown style:

"Obey projects' character line limit wherever possible. Long URLs and tables are the usual suspects when breaking the rule. (Headings also can't be wrapped, but we encourage keeping them short). Otherwise, wrap your text:..."

Reply:
ANTLR tool -> the ANTLR tool

Reply:
Done